nextjs-interface/src/app/ui/dashboard/ChartElement.tsx

```diff
@@ -10,16 +9,29 @@ import {
   ReferenceLine,
   ResponsiveContainer,
 } from "recharts";
-
-import React, { useContext } from "react";
 import { avgData } from "@/lib/context";
 import { ThemeContext } from "@/lib/Theme";
 
 export function ChartElement({ data }: { data: avgData[] }) {
   const { darkMode } = useContext(ThemeContext);
 
-  let textColor;
-  darkMode ? (textColor = "white") : (textColor = "");
+  let textColor = darkMode ? "white" : "";
+
+  if (!Array.isArray(data) || data === undefined) {
+    console.error("Data is not an array or is undefined");
+    return null;
+  }
+
+  const dateChanges = data.reduce((acc: any[], curr, index, src) => {
+    if (
+      index > 0 &&
+      new Date(curr.date).getDate() !== new Date(src[index - 1].date).getDate()
```

Comment:
can't ``new Date(src[index - 1].date).getDate()`` be stored in a variable that can be updated when the date change so we don't recalculate it every time